### PR TITLE
AbstractRightDropdown front URL

### PR DESCRIPTION
**Pull request by AdrienClairembault**
The ajax front file used to load the values of `AbstractRightDropdown` was directly referencing `ShareDashboardDropdown`.
This was not a problem when `ShareDashboardDropdown` was the only implementation of `AbstractRightDropdown` but this is no longer the case since https://github.com/pluginsGLPI/formcreator/pull/2615.

With these changes, each implementation of `AbstractRightDropdown` must have their own ajax front file where they:
* Include the parent ajax file `getAbstractRightDropdownValue.php` which contains the shared logic
* Do their own additional right checks (user who can't access the dropdown shouldn't ajax the ajax endpoint)
* Call `show_rights_dropdown` with the correct class for their specific implementation

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |

### CODE REVIEW

**Comment by AdrienClairembault** [comment]
Fixed tests, should be OK for review if everything is green :)

**Review by orthagh — CHANGES_REQUESTED** [review]
On core dashboard share modal:

Uncaught Exception Error: Non-static method ShareDashboardDropdown::getAjaxUrl() cannot be called statically in /var/www/html/glpi/10.0.git/src/AbstractRightsDropdown.php at line 92

**Comment by AdrienClairembault** [comment]
Indeed, method should have been static.
Don't know why it didn't trigger an error on my side (probably because PHP7 is less strict).
Fixed 14ff59b62d.

File: ajax/getAbstractRightDropdownValue.php [hunk 1]
```diff
@@ -31,12 +31,14 @@
...
 Session::checkLoginUser();
 
-$search = $_POST['searchText'] ?? "";
-
-echo json_encode(ShareDashboardDropdown::fetchValues($search));
+function show_rights_dropdown(string $class)
+{
+    $search = $_POST['searchText'] ?? "";
+    echo json_encode($class::fetchValues($search));
+}
```
Comment (cedric-anne) [inline ajax/getAbstractRightDropdownValue.php:44]: Logic used in other parts of GLPI is reversed, see `dropdown.common.php`.

To use same kind of logic, you should `include(__DIR__ . '/../inc/includes.php');` in the caller file, then define `$class = ShareDashboardDropdown::class;`, then include current file that execute its code.

Comment (AdrienClairembault) [reply ajax/getAbstractRightDropdownValue.php:44]: Wouldn't this be less convenient as it makes the child code run before the parent ?
Even thought it isn't really a parent/child inheritance situation as it isn't OOP code, it seems more intuitive to run the "parent" code first.

If you consider the rights checks for example, shouldn't you do the common check (is the user logged in ?) before the specific ones ?

File: ajax/getShareDashboardDropdownValue.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+include('getAbstractRightDropdownValue.php');
```
[the remark this answers is not in the document]
Comment (cedric-anne) [reply ajax/getShareDashboardDropdownValue.php:34]: To ensure to not search in `include_path`.

**Review by orthagh — APPROVED** [review]

**Pull request merged**